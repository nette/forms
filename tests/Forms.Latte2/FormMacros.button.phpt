<?php

/**
 * Test: FormMacros.
 */

declare(strict_types=1);

use Nette\Bridges\FormsLatte\FormMacros;
use Nette\Forms\Form;
use Tester\Assert;

require __DIR__ . '/../bootstrap.php';

if (version_compare(Latte\Engine::VERSION, '3', '>')) {
	Tester\Environment::skip('Test for Latte 2');
}


$form = new Form;
$form->addSubmit('send', 'Sign in');

$latte = new Latte\Engine;
FormMacros::install($latte->getCompiler());
$latte->addProvider('uiControl', ['myForm' => $form]);

Assert::matchFile(
	__DIR__ . '/expected/FormMacros.button.php',
	$latte->compile(__DIR__ . '/templates/forms.button.latte'),
);

Assert::matchFile(
	__DIR__ . '/expected/FormMacros.button.html',
	$latte->renderToString(__DIR__ . '/templates/forms.button.latte'),
);
