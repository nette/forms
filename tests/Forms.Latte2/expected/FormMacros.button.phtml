<?php
%A%
		$form = $this->global->formsStack[] = $this->global->uiControl["myForm"] /* line 1 */;
		echo '<form';
		echo Nette\Bridges\FormsLatte\Runtime::renderFormBegin(end($this->global->formsStack), [], false);
		echo '>
	<button';
		$ʟ_input = $_input = end($this->global->formsStack)["send"];
		echo $ʟ_input->getControlPart()->attributes() /* line 2 */;
		echo '>
		description of button
	</button>

	<button';
		$ʟ_input = $_input = end($this->global->formsStack)["send"];
		echo $ʟ_input->getControlPart()->attributes() /* line 6 */;
		echo '></button>

	<button';
		$ʟ_input = $_input = end($this->global->formsStack)["send"];
		echo $ʟ_input->getControlPart()->attributes() /* line 8 */;
		echo '>';
		echo htmlspecialchars($ʟ_input->getCaption()) /* line 8 */;
		echo '</button>
';
		echo Nette\Bridges\FormsLatte\Runtime::renderFormEnd(array_pop($this->global->formsStack), false) /* line 1 */;
		echo '</form>
';
%A%
