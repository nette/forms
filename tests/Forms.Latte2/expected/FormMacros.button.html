<form action="" method="post">
	<button type="submit" name="send" value="Sign in">
		description of button
	</button>

	<button type="submit" name="send" value="Sign in"></button>

	<button type="submit" name="send" value="Sign in">Sign in</button>
%A%
</form>
